BUILDER_DEV.md

```diff
@@ -1,390 +1,275 @@
+
+```
+$ sudo ln -s path/to/your/habitat/repo /src
+```
+
+* Cd into your Habitat repo
+
+```
+$ cd path/to/your/habitat/repo
 ```
 
-`config_jobsrv.toml`
-```toml
-key_dir = "/path/to/bldr-key-pair"
+* Run the provision script
 
-[archive]
-backend = "local"
-local_dir = "/tmp"
 ```
+$ sudo ./support/linux/provision.sh
+```
+
+* After the script completes, source cargo/rust
+```
+$ source $HOME/.cargo/env
+```
+
+* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
 
-`config_sessionsrv.toml`
-```toml
-[permissions]
-admin_team = 1995301
-build_worker_teams = [2555389]
-early_access_teams = [1995301]
 ```
+$ sudo make build-srv
+```
+
+* Open up the builder-worker file
+
+```
+$ sudo vim /hab/svc/builder-worker
+```
+
+* Add your Github Auth Key:
+
+**/hab/svc/builder-worker/config.toml**
+```
+auth_token = ""
+```
+
+Should become:
 
-`config_worker.toml`
-```toml
+**/hab/svc/builder-worker/config.toml**
+```
 auth_token = "<your github token>"
-bldr_url = "http://localhost:9636"
-auto_publish = true
-airlock_enabled = false
 ```
 
+* Run this command (it may take awhile) to start all the Builder services
+
+```
+$ sudo -E make bldr-run
+```
 
-(Note: If you want your log files to persist across restarts of your development machine, replace `/tmp` with some other directory. It *must* exist and be writable before you start the job server).
+### UI Setup
 
-Now, modify the `Procfile` (located in your hab repo in the `support` folder) to point the api, sessionsrv, jobsrv and worker services to the previously created config files, e.g.
+* Open a separate shell on your VM
+* Install npm
 
 ```
-api: target/debug/bldr-api start --config /home/your_alias/habitat/config_api.toml
-sessionsrv: target/debug/bldr-sessionsrv start --config /home/your_alias/habitat/config_sessionsrv.toml
-worker: target/debug/bldr-worker start --config /home/your_alias/habitat/config_worker.toml
-jobsrv: target/debug/bldr-jobsrv start --config /home/your_alias/habitat/config_jobsrv.toml
+$ sudo apt-get install npm
+```
+
+* Cd into the builder-web component
+
+```
+$ cd path/to/your/habitat/repo/components/builder-web
+```
+
+* Install all dependencies
+
 ```
+$ npm install
+```
+
+* Create the habitat.conf.js file
+
+```
+$ cp habitat.conf.sample.js habitat.conf.js
+```
+
+* Start the UI
+
+```
+$ sudo npm start
+```
+
+* Open up a browser and navigate to http://localhost:3000/#/pkgs - you should see the Habitat UI running locally.
 
-## Build the Builder services for the first time
-1. Open a new terminal window.
-1. Run `make build-srv`
+### Creating an Origin
 
-Note: this only needs to be done the first time to get the pieces in place. If you clean the project, make sure to unset the variables show below when you do a clean build, otherwise the build will look for packages on your local build service!
+* Navigate to http://localhost:3000/#/pkgs
+* Log in
```

Comment:
There may be a step here to accept the Habitat Builder Dev app in your github account

Reply:
Also, it's important that they grant permissions carefully- if the permissions to repos that are granted are too restrictive, they will not be able to link plans (aka create projects) on the repos that are not accessible to the githab app later.